### DISCUSSION TITLE
core.remote:bastion_host

### DISCUSSION THREAD

**Discussion by komalphadatare** [StackStorm/st2#5526] [Q&A]
Hi,
Jump server: Public IP
Target VM: Private IP
Using same ssh key, we want to login to target env via jump server.
To meet this requirement, we are using core.remote action and configured as below.
cmd: hostname
hosts: <Target VM>
baston_host: <Jump server>
username: ubuntu
password: Empty
private_key = <private_key file path>/<copied private_key file content>
NOTE: private_key file is in stackstom server

We are encountering error as below

Unable to connect to any one of the hosts: [u'192.168.10.207'].

 connect_errors={
  \"192.168.10.207\": {
    \"failed\": true, 
    \"traceback\": \"Traceback (most recent call last):\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/runners/parallel_ssh.py\\\", line 258, in _connect\
    client.connect()\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/runners/paramiko_ssh.py\\\", line 135, in connect\
    self.bastion_client = self._connect(host=self.bastion_host)\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/runners/paramiko_ssh.py\\\", line 689, in _connect\
    passphrase=self.passphrase)\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/runners/paramiko_ssh.py\\\", line 600, in _get_pkey_object\
    raise paramiko.ssh_exception.SSHException(msg)\
SSHException: Invalid or unsupported key type\
\", 
    \"timeout\": false, 
    \"succeeded\": false, 
    \"stdout\": \"\", 
    \"stderr\": \"\", 
    \"error\": \"Connection error. Invalid or unsupported key type\", 
    \"return_code\": 255
  }
}"

**Comment by arm4b** [comment]
And so based on the error message, what's the SSH key type you're using?
Also, what's the st2 version?

**Reply by komalphadatare** [reply]
Hi,

We tried on both versions listed below :
- st2 3.1.0, on Python 2.7.5 
-st2 3.6.0, on Python 3.6.8

Keytype: private key

**Reply by arm4b** [reply]
I'd recommend debugging if the private key type is correct (RSA, DSA, ECDSA, etc) and if you could SSH by using that key manually.

**Reply by komalphadatare** [reply]
Hi, 

Issue Got Resolved .
Thanks for the Support.

**Reply by arm4b** [reply]
Thanks for confirming.
For the history and maybe to help others in the future experiencing similar problem, what was the root cause?

**Reply by komalphadatare** [reply]
Hi,
     When I was passing Private Key material In **core.remote:Private key** Section It was failing Execution. 
Solutions Worked for me listed Below :
   -  Provide Private key File path in core.remote Action which is stored in Stackstorm server.
   - In Workflow define parameter for private key and at runtime pass the private key.

**Reply by komalphadatare** [reply]
Hi,
We want to connect Target VM via Bastion Using Different Private Key and User . How we can Achieve this?
